client-misk/src/main/kotlin/app/cash/backfila/client/misk/BackfillParameters.kt

```diff
@@ -0,0 +1,79 @@
+package app.cash.backfila.client.misk
+
+import app.cash.backfila.protos.service.Parameter
+import com.google.inject.TypeLiteral
+import com.squareup.moshi.Types
+import java.lang.reflect.ParameterizedType
+import kotlin.reflect.KClass
+import kotlin.reflect.KParameter
+import kotlin.reflect.full.memberProperties
+import kotlin.reflect.full.primaryConstructor
+import kotlin.reflect.jvm.jvmErasure
+import misk.exceptions.BadRequestException
+
+interface Parameterized {
+  /**
+   * Returns this backfill's available parameters. These are sent to the Backfila service which will
+   * offer fields for users to type values in for each parameter.
+   *
+   * This can be configured directly but you probably want [DataClassParameter]
+   */
+  val parameters: List<Parameter>
+    get() = listOf()
```

Comment:
default implementations on interfaces feels like a hazard

Reply:
I think in this case it is ok? By default there are no parameters.